mauricekastelijn/XrayViewer
# Initial merge develop into main

- Initial commit - working XrayViewer
- Fix copy of files
- Update packages
- Add prev test
- Fix prev test
- Introduce RunControl
- Use RunControl in XrayViewer
- Factor out RunControl
- Add dedicated RunControl test suite
- Add emptyTest + simplify RunControl
- Simplify ModelXray
- Fix emptyTest + add overflowTest
- Fix overFlowTest
- Cleanup
- Add setImageIndex tests
- Fix bug - test passes
- MVC component version of XrayViewer
- XrayPlayer: Update views from controller through data model. User interfaces for external XrayRun model. Split shared, views, and integration parts.
- Observe Models through interfaces. Create Builder hierarchy separate from controllers. Allows mocking of individual models and controllers.
- Simplify Builders
- Views observe model directly. Combine controller callback and view interfaces into single interface towards Views.
- Controllers add/remove observers of models dynamically upon activation/deactivation. Simplified views and their interface t controllers.
- Re-organize files to make sharing more explicit
- Create README.md
- Fix change event
